### DISCUSSION TITLE
Use quartoDataDir for TinyTeX

### DISCUSSION THREAD

**Discussion by averms** [quarto-dev/quarto-cli#11748] [Feature Requests]
### Description

On Linux, Quarto already follows the XDG Base Directory spec, which is very nice. It results in a clean $HOME. However, the one place it doesn't is when installing TinyTeX. It would be great if TinyTeX was also installed following the XDG Base Directory spec just like everything else. This would put it in XDG_DATA_HOME/quarto, right next to chromium.

I understand that upstream TinyTeX already has defaults and there is value in sticking to them. However, I noticed that you guys don't even use the upstream script and don't add the binaries to the user's PATH so you're already deviating from upstream.

**Comment by cscheid** [comment]
@cderv Is this something we could reasonably do?

**Reply by cderv** [reply]
I think this is something we can do, in the sense that technically we can unzip into another directory. 

However, we do use today the same default path as upstream TinyTeX install script so that we don't install a second time TinyTeX if already available. This means that if a user already installed TinyTeX another way, this version will be found and used by Quarto even if not on PATH. To my knowledge, that is the reason for where the distribution is installed by default by Quarto - so that it works with other tools assuming TinyTeX there, and for Quarto to found it too. 

So I don't know if we could reasonably change the default. It would work ok for Quarto itself, but it would have impact on some situation assuming `~/.TinyTeX` - what do you think ? 

Though, we could add a flag to install TinyTeX elsewhere if desired. 

Anyhow, if we do that for Linux, should we also do it for other OS ? It would mean 

- Linux: `~/.TinyTeX` would be now `~/.local/share/TinyTeX`
- Darwin: `~/Library/TinyTeX` would be now `~/Library/Application Support/TinyTeX`
- Windows: `%APPDATA%/TinyTeX` which would not change I think (or maybe `%LOCALAPPDATA%` ?)

@yihui do you have any insight on changing install path for TinyTeX ? From your experience, could it have some impact ?

**Reply by yihui** [reply]
I think it should be fine to change the default installation path, as long as Quarto is still able to find other instances of TinyTeX in the system.

In TinyTeX's installation scripts, users can define the custom installation path via the env var [`TINYTEX_DIR`](https://github.com/rstudio/tinytex/blob/f9205b45c5b6bdfea9a9af7d9395a2a76205139b/tools/install-bin-unix.sh#L28-L30). If users want to install to `XDG_DATA_HOME`, they can set `TINYTEX_DIR=$XDG_DATA_HOME` before running the installation script.

**Reply by cderv** [reply]
> I think it should be fine to change the default installation path, as long as Quarto is still able to find other instances of TinyTeX in the system.

Great thanks ! 

> In TinyTeX's installation scripts, users can define the custom installation path

I opened 
- https://github.com/quarto-dev/quarto-cli/issues/11800

to track the support of custom install path with Quarto like with R or other install script.
